src/css.js

```diff
@@ -257,7 +257,9 @@ jQuery.extend( {
 			if ( !hooks || !( "set" in hooks ) ||
 				( value = hooks.set( elem, value, extra ) ) !== undefined ) {
 
-				style[ name ] = value;
+				if ( support.styleWhitespace || !/^\s+$/.test( value ) ) {
+					style[ name ] = value;
+				}
 			}
```

Comment:
+1 for without support test approach, since that should be smaller right and it shouldn't have affect the perf it seems